smcontroller/smcontroller.go

```diff
@@ -99,396 +101,241 @@ type CertificateProvider interface {
 // New creates new SM controller.
 func New(
 	cfg *config.Config, messageSender MessageSender, alertSender AlertSender, monitoringSender MonitoringSender,
-	urlTranslator URLTranslator, certProvider CertificateProvider, cryptcoxontext *cryptutils.CryptoContext,
+	certProvider CertificateProvider, cryptcoxontext *cryptutils.CryptoContext,
 	insecureConn bool,
 ) (controller *Controller, err error) {
 	log.Debug("Create SM controller")
 
 	controller = &Controller{
+		eventChannel:              make(chan smCtrlInternalMsg, 10),
 		messageSender:             messageSender,
 		alertSender:               alertSender,
 		monitoringSender:          monitoringSender,
-		urlTranslator:             urlTranslator,
+		runInstancesStatusChan:    make(chan launcher.NodeRunInstanceStatus, statusChanSize),
 		updateInstancesStatusChan: make(chan []cloudprotocol.InstanceStatus, statusChanSize),
-		runInstancesStatusChan:    make(chan unitstatushandler.RunInstancesStatus, statusChanSize),
-		clients:                   make(map[string]*smClient),
+		nodes:                     make(map[string]*smHandler),
+	}
+
+	for _, nodeID := range cfg.SMController.NodeIDs {
+		controller.nodes[nodeID] = nil
+	}
+
+	if controller.smSerever, err = newSmServer(cfg, messageSender, alertSender, monitoringSender, controller.eventChannel,
+		controller.runInstancesStatusChan, controller.updateInstancesStatusChan,
+		certProvider, cryptcoxontext, insecureConn); err != nil {
+		return nil, aoserrors.Wrap(err)
 	}
 
 	ctx, cancelFunction := context.WithCancel(context.Background())
 	controller.cancelFunction = cancelFunction
 
-	defer func() {
-		if err != nil {
-			controller.Close()
-			controller = nil
-		}
-	}()
-
-	var secureOpt grpc.DialOption
+	controller.connectionTimer = time.NewTimer(time.Second * 1)
```

Comment:
Add const for connection timer and one second is not enough.

Reply:
done